enterprise/com/src/main/java/org/neo4j/com/IdleChannelReaper.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.com;
+
+import org.jboss.netty.channel.Channel;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.neo4j.helpers.Clock;
+import org.neo4j.logging.Log;
+import org.neo4j.logging.LogProvider;
+
+/**
+ * Keeps track of a set of channels and when they were last active.
+ * Run this object periodically to close any channels that have been inactive for longer than the threshold.
+ * This functionality is required because sometimes Netty doesn't tell us when channels are
+ * closed or disconnected. Most of the time it does, but this acts as a safety
+ * net for those we don't get notifications for. When the bug is fixed remove this class.
+ */
+public class IdleChannelReaper implements Runnable
+{
+    private final Map<Channel,Request> connectedChannels = new ConcurrentHashMap<>();
```

Comment:
There is no need for a `ConcurrentHashMap` here, all the access of the map are externally `synchronized`.